python/pyarrow/tests/test_dataset.py

```diff
@@ -16,31 +16,29 @@
 # under the License.
 
 import contextlib
-import os
-import posixpath
 import datetime
+import os
 import pathlib
 import pickle
+import posixpath
 import sys
-import textwrap
 import tempfile
+import textwrap
 import threading
 import time
-
 from urllib.parse import quote
 
 import numpy as np
-import pytest
-
 import pyarrow as pa
 import pyarrow.compute as pc
 import pyarrow.csv
-import pyarrow.json
 import pyarrow.feather
 import pyarrow.fs as fs
-from pyarrow.tests.util import (change_cwd, _filesystem_uri,
-                                FSProtocolClass, ProxyHandler,
-                                _configure_s3_limited_user)
+import pyarrow.json
+import pytest
+from pyarrow.tests.util import (FSProtocolClass, ProxyHandler,
+                                _configure_s3_limited_user, _filesystem_uri,
+                                change_cwd)
```

Comment:
I ran `archery lint --python --fix` which appears to have shuffled around a bit bits in this file that I didn't change.

Reply:
It seems like archery no longer re-orders imports, so I guess that must have been changed since I originally made the PR. I suspect the mis-ordering was related to archery changes that where not rebased correctly. 

I put `pytest` back where its supposed to be but I think `pyarrow.json` is actually correct to me.